Remove references to scxa_tsne table, use base container, M1 partly native

pmb59 opened the pull request:
This updates tests to changes in schema versions, most notably the addition of scxa_coords and removal of tsne tables.

This also switches the testing to use a base container created the miniconda template approach, which happens to use a user different from root, which has some privs implications.

pmb59 committed:
See https://github.com/ebi-gene-expression-group/atlas-schemas/pull/23#issuecomment-891842044

pmb59 committed:
Removes references to (currently unused) scxa_tsne

pmb59 committed:
Removes references to (currently unused) table scxa_tsne

pinin4fjords commented:
@pmb59 You'll need to update the tests as well to satisfy the CI I think. 

Also, please add something descriptive to the description for future-us to read.

pinin4fjords commented:
Also, could you update the README?

pmb59 committed:
Removes @tests related to the deprecated scxa_tsne table:
"TSNE: Load data”
"Dimred parameters: check that JSON queries run and return expected values”
"TSNE: Check number of loaded rows”
"TSNE: Delete experiment”

pmb59 commented:
I have modified the tests but the CI failure looks related to Docker container.

(btw , probably including tests for `scxa_coords` to be done at a later stage)

pmb59 committed:
Updates the section 'scxa_coords Table' in the README file to remove references to scxa_tsne and include reference to UMAP files and their expected prefixes.

pcm32 commented on bin/reindex_tables.sh:
```diff
@@ -2,12 +2,12 @@
 REINDEX TABLE experiment;
 
-CLUSTER scxa_tsne USING scxa_tsne_experiment_accession_cell_id_perplexity_pk;
+#CLUSTER scxa_tsne USING scxa_tsne_experiment_accession_cell_id_perplexity_pk;
```
```suggestion
```
I would just remove this.

pcm32 reviewed (commented):
I think that we are lacking here to point to the latest develop version of atlas-schemas sub-module.

pmb59 committed:
Update bin/reindex_tables.sh

Co-authored-by: Pablo Moreno <pcm32@users.noreply.github.com>

pcm32 committed:
Merge branch 'develop' into feature/rm-references-tsne-table

pcm32 committed:
Change base until this is replaced

pcm32 committed:
micromamba cannot set channels, try newer miniconda

pcm32 committed:
Use base container and update atlas-schemas ref

pcm32 committed:
Set scratch dir to /tmp for micromamba compat

pcm32 committed:
More permissions

pcm32 committed:
Try with all writable

pcm32 committed:
Missing run

pcm32 committed:
Use scratch dir

pcm32 committed:
Use scratch

pcm32 committed:
Separate tests to isolate failure

pcm32 committed:
Simplify copies to writable directory

pcm32 committed:
Fix copy

pcm32 committed:
Passing tests

pcm32 committed:
Missing reindex!

pinin4fjords commented:
As requested by past-me, can future-me get a description please?

pcm32 commented:
I have added a description above.

pinin4fjords commented on tests/random-data-set.bats:
```diff
@@ -262,46 +272,12 @@
   [ "$status" -eq 0 ]
 }
 
-@test "TSNE: Load data" {
-  export EXP_ID=TEST-EXP1
-  run load_db_scxa_dimred.sh
-  echo "output = ${output}"
-  [ "$status" -eq 0 ]
-}
-
-@test "Dimred parameters: check that JSON queries run and return expected values" {
-    target_perps="1 5 10 15 20  "
-    perps=$(echo "select distinct parameterisation->0->'perplexity' as perplexity from scxa_coords order by perplexity" | psql -At $dbConnection | tr '\n' ' ')
-    run [ "$perps" = "$target_perps" ]
-}
-
-@test "TSNE: Check number of loaded rows" {
-  # Get third line with count of total entries in the database after our load
-  count=$(echo "SELECT COUNT(*) FROM scxa_tsne" | psql -v ON_ERROR_STOP=1 $dbConnection | awk 'NR==3')
-  # TODO improve, highly dependent on test files we have, but in a hurry for now.
-  run [ $count -eq 250 ]
-  echo "output = ${output}"
-  [ "$status" -eq 0 ]
-}
-
-@test "TSNE: Delete experiment" {
-  export EXP_ID=TEST-EXP1
-  run delete_db_scxa_dimred.sh
-  echo "output = ${output}"
-  [ "$status" -eq 0 ]
-  count=$(echo "SELECT COUNT(*) FROM scxa_tsne WHERE experiment_accession = '"$EXP_ID"'" | psql -v ON_ERROR_STOP=1 $dbConnection | awk 'NR==3')
```
Why are we losing all these tests? One possibly needs adapting for the new table, but all the tests seem like they should stay in some form unless theres a good reason I'm missing to strip them.

pcm32 replied:
@pmb59 were these tests replaced by others that would be equivalent for the scxa coords? I only took part of fixing the container and tests parts.

pcm32 replied:
If not, could you add them now that the tests are passing with the new container setup please? Thanks.

pcm32 replied:
Ok, I think I can do this quickly.

pcm32 committed:
Not needed path

pcm32 committed:
Re-instate tests, loading apparently not working

pcm32 committed:
Should fix tests

pcm32 committed:
This shoudl really fix them..

pcm32 committed:
Use scratch dir in dimred

pinin4fjords reviewed (approved):
Looks good- thank you for sorting those last bits @pcm32 !
